src/block/block_ext.c

```diff
@@ -74,11 +79,20 @@ __block_off_srch(WT_EXT **head, wt_off_t off, WT_EXT ***stack, bool skip_off)
      * the skip_off flag is set, offset the skiplist array by the depth specified in this particular
      * structure.
      */
-    for (i = WT_SKIP_MAXDEPTH - 1, extp = &head[i]; i >= 0;)
-        if (*extp != NULL && (*extp)->off < off)
+    ext_tmp = NULL;
+    for (i = WT_SKIP_MAXDEPTH - 1, extp = &head[i]; i >= 0;) {
+        if (*extp != NULL && (*extp)->off < off) {
+            ext_tmp = *extp;
             extp = &(*extp)->next[i + (skip_off ? (*extp)->depth : 0)];
-        else
+        } else {
             stack[i--] = extp--;
```

Comment:
Do we have to update the new `ext_tmp` in the `else` statement?

Reply:
The else branch only retrieves the position by the off that be searched at each level in the skip list. I think that the else branch does not require an ext_tmp operation.

Reply:
The else branch only retrieves the position of the offset to be searched for at each level in the skip list. I believe that the else branch does not require an ext_tmp operation.

Reply:
I'm quite unfamiliar with the code myself, so I will need to carefully review this file to understand how things come together. Please expect some delay as I cannot actively work on this.

Reply:
got it, thanks.